src/tests/end2end/D3D12ResidencyManagerTests.cpp

```diff
@@ -134,31 +221,25 @@ TEST_F(D3D12ResidencyManagerTests, CreateResourceHeap) {
     // Assume tier 1, which all adapters support.
     heapDesc.Flags |= D3D12_HEAP_FLAG_ALLOW_ONLY_BUFFERS;
 
-    auto createHeapFn = [&](ID3D12Pageable** ppPageableOut) -> HRESULT {
-        ComPtr<ID3D12Heap> heap;
-        if (FAILED(mDevice->CreateHeap(&heapDesc, IID_PPV_ARGS(&heap)))) {
-            return E_FAIL;
-        }
-        *ppPageableOut = heap.Detach();
-        return S_OK;
-    };
-
-    auto badCreateHeapFn = [&](ID3D12Pageable** ppPageableOut) -> HRESULT {
-        // No pageable set should E_FAIL.
-        return S_OK;
-    };
+    CreateHeapFnContext createHeapFnContext(mDevice.Get(), heapDesc);
+    BadCreateHeapFnContext badCreateHeapFnContext;
 
     HEAP_DESC resourceHeapDesc = {};
     resourceHeapDesc.SizeInBytes = kHeapSize;
     resourceHeapDesc.MemorySegmentGroup = DXGI_MEMORY_SEGMENT_GROUP_LOCAL;
 
-    ASSERT_FAILED(CreateHeap(resourceHeapDesc, residencyManager.Get(), badCreateHeapFn, nullptr));
+    ASSERT_FAILED(CreateHeap(resourceHeapDesc, residencyManager.Get(),
+                             BadCreateHeapFnContext::CreateHeapWrapper,
+                             reinterpret_cast<void*>(&badCreateHeapFnContext), nullptr));
 
-    ASSERT_SUCCEEDED(CreateHeap(resourceHeapDesc, residencyManager.Get(), createHeapFn, nullptr));
+    ASSERT_SUCCEEDED(CreateHeap(resourceHeapDesc, residencyManager.Get(),
+                                CreateHeapFnContext::CreateHeapWrapper,
+                                reinterpret_cast<void*>(&createHeapFnContext), nullptr));
 
     // Create a resource heap without residency.
     ComPtr<IHeap> resourceHeap;
-    ASSERT_SUCCEEDED(CreateHeap(resourceHeapDesc, nullptr, createHeapFn, &resourceHeap));
+    ASSERT_SUCCEEDED(CreateHeap(resourceHeapDesc, nullptr, CreateHeapFnContext::CreateHeapWrapper,
+                                reinterpret_cast<void*>(&createHeapFnContext), &resourceHeap));
```

Comment:
nit: could we we not just pass `&wrapper` directly?

Reply:
If by directly you mean without the cast, yes. Fixed.

Reply:
Done.